### PR TITLE
Parse abc files

### PR DESCRIPTION
**Pull request by felix-hauger**
Parse .abc files and return an array of Instruction struct.

Each Instruction struct contains the instruction, potentials arguments and the total size of all elements of the struct.

The size just is a placeholder for now and depending on how we calculate it, it might not be needed in the Instruction struct and if so may be deleted later.

### CODE REVIEW

**Comment by felix-hauger** [comment]
Seems okay, need to create clean functions to write bytecode

File: src/TestOperation.c [hunk 1]
```diff
@@ -46,6 +46,8 @@ void writeHeader(int filedes, int programSize, char programName[PROGRAM_NAME_SIZ
 {
     write(filedes, "CODE", 4);
 
+    // write(filedes, &programSize, 4);
```
Comment (felix-hauger) [inline src/TestOperation.c]: remove useless comment

File: src/TestOperation.c [hunk 2]
```diff
@@ -46,6 +46,8 @@ void writeHeader(int filedes, int programSize, char programName[PROGRAM_NAME_SIZ
...
+
     writeIntegerForCompile(filedes, programSize);
     // write(filedes, &(bswap_32(programSize)), sizeof(programSize));
```
Comment (felix-hauger) [inline src/TestOperation.c]: Remove useless commented code

File: src/main.c [hunk 3]
```diff
@@ -13,31 +13,46 @@ int main(int argc, char **argv)
 
     char *programName = "abcdefghijklmnop";
 
-    int result = getFunctionIdFromName("ADD");
+
+
+    // int result = getFunctionIdFromName("ADD");
```
Comment (felix-hauger) [inline src/main.c]: remove useless commented code

File: src/main.c [hunk 4]
```diff
@@ -13,31 +13,46 @@ int main(int argc, char **argv)
...
     // int reversed = bswap_32(result);
```
Comment (felix-hauger) [inline src/main.c]: remove useless commented code

File: src/main.c [hunk 5]
```diff
@@ -13,31 +13,46 @@ int main(int argc, char **argv)
...
-    char buffer[11] = {0}; // Initialize buffer with 11 zero bytes
-    write(1, buffer, 11);   // Write 11 bytes of zero to stdout
+    // char buffer[11] = {0}; // Initialize buffer with 11 zero bytes
+    // write(1, buffer, 11);   // Write 11 bytes of zero to stdout
 
-    printf("Result: %d\n", result);
+    // printf("Result: %d\n", result);
     //result = bswap_32(result)
 
     // write(filedes, &reversed , 4);
```
Comment (felix-hauger) [inline src/main.c]: remove useless commented code

**Review by felix-hauger — COMMENTED** [review]
Looks ok, nice added comment for the header

**Review by felix-hauger — COMMENTED** [review]
ok

**Review by driss-khelfi — APPROVED** [review]

**Pull request merged**